firedrake/slate/slac.py

```diff
@@ -0,0 +1,529 @@
+"""This is SLATE's Linear Algebra Compiler (SLAC). This module is responsible
+for generating C++ kernel functions representing symbolic linear algebra
+expressions written in SLATE.
+
+STAC uses both Firedrake's form compiler, the Two-Step Form Compiler (TSFC)
+and COFFEE's kernel abstract syntax tree (AST) optimizer. TSFC provides SLAC with
+appropriate kernel functions (in C) for evaluating integral expressions (finite
+element variational forms written in UFL). COFFEE's AST optimizing framework
+produces the resulting kernel AST returned by: `compile_slate_expression`
+
+The Eigen C++ library (http://eigen.tuxfamily.org/) is required, as all low-level
+numerical linear algebra operations are performed using the `Eigen::Matrix` methods
+built into Eigen.
+"""
+
+import sys
+import firedrake
+import tsfc
+import operator
+
+from coffee import base as ast
+from coffee.visitor import Visitor
+
+from firedrake.tsfc_interface import SplitKernel, KernelInfo
+
+from slate import *
```

Comment:
I recommend against importing star, flake8 is much more useful if you don't do that.